src/core/connection_pool.c

```diff
@@ -0,0 +1,350 @@
+/*++
+
+    Copyright (c) Microsoft Corporation.
+    Licensed under the MIT License.
+
+Abstract:
+
+    Definitions for the MsQuic Connection Pool API, which allows clients to
+    create a pool of connections that are spread across RSS cores.
+
+--*/
+
+#include "precomp.h"
+
+#ifdef QUIC_CLOG
+#include "connection_pool.c.clog.h"
+#endif
+
+_IRQL_requires_max_(PASSIVE_LEVEL)
+QUIC_STATUS
+QUIC_API
+MsQuicConnectionPoolCreate(
+    _In_ HQUIC Registration,
+    _In_ QUIC_CONNECTION_CALLBACK_HANDLER Handler,
+    _In_opt_ void* Context,
+    _In_opt_ const char* ServerName,
```

Comment:
The ConnectionStart api takes only a "ServerName" (which can be an IP literal).
Should we stay consistent and have a single mandatory parameter instead of two optional ones? It would leave less room for error.

Reply:
I agree. I've changed the API to take a configuration struct, and the ServerName is a required parameter.